pkg/providers/vmware/client/client_test.go

```diff
@@ -0,0 +1,155 @@
+package client_test
+
+import (
+	"github.com/kubevirt/vm-import-operator/pkg/providers/vmware/client"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/simulator"
+)
+
+var _ = Describe("Test VMware rich client with vCenter", func() {
+	var (
+		model               *simulator.Model
+		server              *simulator.Server
+		vm                  *simulator.VirtualMachine
+		username            string
+		password            string
+		simulatorThumbprint = ""
+	)
+
+	BeforeEach(func() {
+		model = simulator.VPX()
+		_ = model.Create()
+		server = model.Service.NewServer()
+		vm = simulator.Map.Any("VirtualMachine").(*simulator.VirtualMachine)
+		username = server.URL.User.Username()
+		password, _ = server.URL.User.Password()
+	})
+
+	AfterEach(func() {
+		model.Remove()
+		server.Close()
+	})
+
+	It("should connect to vCenter", func() {
+		_, err := client.NewRichVMWareClient(server.URL.String(), username, password, simulatorThumbprint)
+		Expect(err).To(BeNil())
+	})
+
+	It("should retrieve a VM by ID", func() {
+		richClient, err := client.NewRichVMWareClient(server.URL.String(), username, password, simulatorThumbprint)
+		Expect(err).To(BeNil())
+		vmRef := vm.Reference().Value
+		rawVm, err := richClient.GetVM(&vmRef, nil, nil, nil)
```

Comment:
Please add some "air" to the test cases by separating "given", "when" and "then" parts; for example:

```go
		richClient, err := client.NewRichVMWareClient(server.URL.String(), username, password, simulatorThumbprint)
		Expect(err).To(BeNil())
		vmRef := vm.Reference().Value

		rawVm, err := richClient.GetVM(&vmRef, nil, nil, nil)

		Expect(err).To(BeNil())
		retrievedVm, ok := rawVm.(*object.VirtualMachine)
		Expect(ok).To(BeTrue())
		Expect(retrievedVm.Reference()).To(Equal(vm.Reference()))
```